### PR TITLE
Add clangd support for our C++ code.

### PR DESCRIPTION
**Pull request by vext01**
Jump to definition now works reliably, even into LLVM headers.

### CODE REVIEW

File: ykllvmwrap/build.rs [hunk 1]
```diff
@@ -38,8 +80,19 @@ fn main() {
     if env::var("PROFILE").unwrap() == "release" {
         comp.flag("-DNDEBUG");
     }
+
+    // We use a compiler wrapper to capture compilation commands and generate a
+    // `compile_commands.json` database for clangd.
+    let td = TempDir::new().unwrap();
+    env::set_var("YKLLVMWRAP_JSON_TEMP", td.path());
```
Comment (ltratt) [inline ykllvmwrap/build.rs:87]: I guess there's no way to do this *just* for the ykllvmwrap subprocess?

Comment (vext01) [reply ykllvmwrap/build.rs:87]: Afraid not. I had hoped for a `compiler.env(key, val)`, but there isn't one.

**Comment by ltratt** [comment]
OK with me. I'll leave this one to @ptersilie.

**Pull request merged**